libraries/core_libs/node/src/node.cpp

```diff
@@ -118,6 +132,29 @@ void FullNode::init() {
                                        next_votes_mgr_, dag_mgr_, trx_mgr_);
 }
 
+void FullNode::setupMetricsUpdaters() {
+  auto network_metrics = metrics_->getMetrics<metrics::NetworkMetrics>();
+  network_metrics->setPeersCountUpdater([&network = network_]() { return network->getPeerCount(); });
```

Comment:
I think in general it is better practice to use copy here -> network = network_ due to potential dangling reference and 1 time copy of pointer is cheap anyway